solver/src/encoding.rs

```diff
@@ -37,6 +39,51 @@ fn encode_order_flags(order: &UserOrder) -> u8 {
     result
 }
 
+fn encode_uniswap_call(
+    amount_in: U256,
+    amount_out_min: U256,
+    token_in: H160,
+    token_out: H160,
+    payout_to: H160,
+) -> [u8; UNISWAP_INTERACTION_SIZE] {
+    let uniswap = IUniswapV2Router02::at(&dummy::dummy_web3(), H160::zero());
+    let method = uniswap.swap_exact_tokens_for_tokens(
+        amount_in,
+        amount_out_min,
+        vec![token_in, token_out],
+        payout_to,
+        U256::MAX,
+    );
+    let data = method.tx.data.unwrap().0;
+    let mut result = [0u8; UNISWAP_INTERACTION_SIZE];
+    result.copy_from_slice(data.as_slice());
+    result
```

Comment:
nit: We can use a cool new Rust feature: https://doc.rust-lang.org/std/convert/trait.TryFrom.html#impl-TryFrom%3CVec%3CT%3E%3E
```suggestion
    result.try_into().expect("encoded Uniswap interaction unexpected byte length")
```